### PR TITLE
Refactor Interpreter

### PR DESCRIPTION
**Pull request by galpHub**
### Feature

* Code Refactoring

### Description

* Extract signature checkers
* Extract stack management
* Simplify script evaluation

### Notes

* No additional notes at this time

### CODE REVIEW

File: divi/src/script/opcodes.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    case OP_NOP4                   : return "OP_NOP4";
+    case OP_NOP5                   : return "OP_NOP5";
+    case OP_NOP6                   : return "OP_NOP6";
+    case OP_NOP7                   : return "OP_NOP7";
+    case OP_NOP8                   : return "OP_NOP8";
+    case OP_NOP9                   : return "OP_NOP9";
+    case OP_NOP10                  : return "OP_NOP10";
+
+    // zerocoin
+    case OP_ZEROCOINMINT           : return "OP_ZEROCOINMINT";
```
Comment (99darwin) [inline divi/src/script/opcodes.cpp:135]: We really don't use Zerocoin, so these opcodes are probably removable

Comment (galpHub) [reply divi/src/script/opcodes.cpp:135]: You're right. I don't think I saw them in the scripting language either. However, removing these opcodes need to be removed from the wallet.cpp and script.cpp file, along with all their references. Do you think you want those changes as part of this PR or should I open one after?

Comment (99darwin) [reply divi/src/script/opcodes.cpp:135]: It's not a dire situation and doesn't affect this current PR. For that reason, I think we can backlog the task and move forward with this merge.

**Comment by galpHub** [comment]
Ok, sure. That just might be tricky because of the myriad of places that it touches.

**Pull request merged**